connectivity/check/action.go

```diff
@@ -942,3 +977,68 @@ func (a *Action) validateFlowsForPeer(ctx context.Context, reqs []filters.FlowSe
 		}
 	}
 }
+func (a *Action) GetEgressMetricsRequirements() []metricsResult {
+	return a.expEgress.Metrics
+}
+func (a *Action) GetIngressMetricsRequirements() []metricsResult {
+	return a.expIngress.Metrics
+}
+
+const (
+	metricsCollectionTimeout      = time.Second * 30
+	metricCollectionIntervalRetry = time.Second * 5
+)
+
+// ValidateMetrics confronts the expected metrics against the last ones retrieves.
+func (a *Action) ValidateMetrics(ctx context.Context, pod Pod, results []metricsResult) {
+	node := pod.NodeName()
+
+	if len(results) == 0 {
+		// Early exit.
+		a.Debugf("no metrics requirements to validate for this node %s", node)
+		return
+	}
+
+	for _, r := range results {
+		a.validateMetric(ctx, node, r)
+	}
+}
+
+func (a *Action) validateMetric(ctx context.Context, node string, result metricsResult) {
+	// ctx is used here as the duration of the retries to collect metrics.
+	ctx, cancel := context.WithTimeout(ctx, metricsCollectionTimeout)
+	defer cancel()
+
+	// ticker here is used as the interval duration between before the next retry.
+	ticker := time.NewTicker(metricCollectionIntervalRetry)
+	defer ticker.Stop()
+
+	for {
+		// Collect the new metrics.
+		newMetrics, err := a.collectPrometheusMetrics(result.Source)
```

Comment:
As an optimization, we could consider adding a `collectPrometheusMetricsForNode` function, that only port-forwards to a particular node

Reply:
Fixed.